spring-cloud-sleuth-instrumentation/src/main/java/org/springframework/cloud/sleuth/instrument/batch/TraceJobExecutionListener.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.sleuth.instrument.batch;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobExecutionListener;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanAndScope;
+import org.springframework.cloud.sleuth.Tracer;
+
+class TraceJobExecutionListener implements JobExecutionListener {
+
+	private final Tracer tracer;
+
+	private static Map<JobExecution, SpanAndScope> SPANS = new ConcurrentHashMap<>();
+
+	TraceJobExecutionListener(Tracer tracer) {
+		this.tracer = tracer;
+	}
+
+	@Override
+	public void beforeJob(JobExecution jobExecution) {
+		Span span = this.tracer.nextSpan().name(jobExecution.getJobInstance().getJobName());
+		span.tag("job.name", jobExecution.getJobInstance().getJobName());
```

Comment:
that's a great question cc @benas

Reply:
Ohh... that's a nice idea.   Except job parameters we will not want to share as they may contain sensitive information.

Reply:
@cppwfs Does it have an id or something that the user can reference? Or does it strictly belong to the execution and the execution id identifies the parameters too?

Reply:
I agree with @jonatan-ivanov that adding a few more tags here in addition to the job name would be helpful from the user's point of view:
* `JobInstanceId` and `JobExecutionId` uniquely identify job executions. I think they should be added as tags to make it possible to distinguish a trace from another when multiple job instances are executed in parallel.
* `JobParameters`: could be added, but to Glenn's point, this could be a security risk if users misuse job parameters and pass sensitive information there (FTR, job parameters are **not** expected to be used for sensitive information as they are persisted in clear text, a bit more context about the matter here: https://github.com/spring-projects/spring-batch/issues/3680). I would be careful about adding them here as users might misuse them for sensitive infos.
* `version`: this is a technical field used internally by Spring Batch for its optimistic locking strategy when interacting with the job repository. I believe this has no added value to the end user and should not be added.
* `ExitStatus` does not make sense here, we are in  `JobExecutionListener#beforeJob`, so the job did not exit yet at this point (and did not even started).

Reply:
👍🏼
My comment is not exclusive to `beforeJob`, I just want to gather what other details can we add to the `Span` that are useful for the users (some of them only makes sense in `afterJob`).
Since there are quite a few things in the batch domain: `Job`, `JobInstance`, `JobExecution`, `JobParameters` (+ `StepExecution` in the other class), do you have anything else in mind that we should add?

Reply:
The list above is a good starting point at the job level.